### PR TITLE
KTLN-746: Guide to Reading From a YAML File in Kotlin

### PR DESCRIPTION
**Pull request by tapankavasthi**
### Summary
Demonstrate reading YAML content in Kotlin

### Jira
[KTLN-746](https://jira.baeldung.com/browse/KTLN-746)

### CODE REVIEW

**Comment by tapankavasthi** [comment]
test this please

**Comment by tapankavasthi** [comment]
test this please

File: kotlin-yaml/src/test/kotlin/com/baeldung/yaml/yamlkt/YamlktUnitTest.kt [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+import net.mamoe.yamlkt.Yaml
+import net.mamoe.yamlkt.YamlMap
+import net.mamoe.yamlkt.YamlNull
+import net.mamoe.yamlkt.getPrimitive
+
+class YamlktUnitTest {
+
+    private val yamlDefault = Yaml.Default
```
Comment (theangrydev) [inline kotlin-yaml/src/test/kotlin/com/baeldung/yaml/yamlkt/YamlktUnitTest.kt:15]: Let's use a fully qualified import to make it clear which library we are importing from

File: kotlin-yaml/src/test/kotlin/com/baeldung/yaml/kaml/KamlUnitTest.kt [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+import com.charleskorn.kaml.*
+import kotlinx.serialization.decodeFromString
+import org.junit.jupiter.api.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+import kotlin.test.assertFalse
+
+class KamlUnitTest {
+
+    private val yamlDefault = Yaml.default
```
Comment (theangrydev) [inline kotlin-yaml/src/test/kotlin/com/baeldung/yaml/kaml/KamlUnitTest.kt:14]: Let's use a fully qualified import to make it clear which library we are importing from

File: kotlin-yaml/src/test/kotlin/com/baeldung/yaml/kaml/KamlUnitTest.kt [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+        val scalarIntYamlStr = "123"
+        val parsedData: YamlScalar = yamlDefault.parseToYamlNode(scalarIntYamlStr).yamlScalar
+        assertEquals(123, parsedData.toInt())
+    }
+
+
+    @Test
+    fun `Return float when parsing scalar float number in a yaml string using decodeFromString`() {
+        val scalarFloatYamlStr = "12.3"
+        val yaml = Yaml.default
```
Comment (theangrydev) [inline kotlin-yaml/src/test/kotlin/com/baeldung/yaml/kaml/KamlUnitTest.kt:48]: Use existing field?

**Review by theangrydev — CHANGES_REQUESTED** [review]
Nice! Just a few minor points

File: kotlin-yaml/src/main/kotlin/com/baeldung/yaml/kaml/Kaml.kt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+fun getUsersUsingYamlMapSerializer(fileContent: String): Users {
+    val yaml = Yaml(configuration = YamlConfiguration(strictMode = false))
+    val result: MutableList<User> = mutableListOf()
+
+    val data: YamlMap = yaml.parseToYamlNode(fileContent).yamlMap
+    val usersYamlList: YamlList = data.get<YamlList>("users")!!.yamlList
+
+
+
```
[the remark this answers is not in the document]
Comment (tapankavasthi) [reply kotlin-yaml/src/main/kotlin/com/baeldung/yaml/kaml/Kaml.kt]: Fixed

File: kotlin-yaml/pom.xml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+                <configuration>
+                    <compilerPlugins>
+                        <plugin>kotlinx-serialization</plugin>
+                    </compilerPlugins>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+
```
[the remark this answers is not in the document]
Comment (tapankavasthi) [reply kotlin-yaml/pom.xml]: Fixed

File: kotlin-yaml/src/test/kotlin/com/baeldung/yaml/yamlkt/YamlktUnitTest.kt (hunk 1 above)
Comment (tapankavasthi) [reply kotlin-yaml/src/test/kotlin/com/baeldung/yaml/yamlkt/YamlktUnitTest.kt]: Fixed

File: kotlin-yaml/src/test/kotlin/com/baeldung/yaml/kaml/KamlUnitTest.kt (hunk 2 above)
Comment (tapankavasthi) [reply kotlin-yaml/src/test/kotlin/com/baeldung/yaml/kaml/KamlUnitTest.kt]: Fixed

File: kotlin-yaml/src/test/kotlin/com/baeldung/yaml/kaml/KamlUnitTest.kt (hunk 3 above)
Comment (tapankavasthi) [reply kotlin-yaml/src/test/kotlin/com/baeldung/yaml/kaml/KamlUnitTest.kt]: Fixed

**Comment by theangrydev** [comment]
@tapankavasthi going forwards can you try not to force push when possible, it's hard to review the diff versus the last time I reviewed the PR if the commit history is lost

**Review by theangrydev — APPROVED** [review]

**Comment by tapankavasthi** [comment]
> @tapankavasthi going forwards can you try not to force push when possible, it's hard to review the diff versus the last time I reviewed the PR if the commit history is lost

sure, @theangrydev. I wanted to keep the commit history clean, so had to squash the commits.

Here is the diff, in case you want to double-check:

https://github.com/Baeldung/kotlin-tutorials/compare/79bafd90587b07ddfe818844b8fe751e55d535a1..4f0897fe0440fa05f4ce74a26385afad0d791a8d

**Pull request merged**